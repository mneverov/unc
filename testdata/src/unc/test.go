package unc

func foo() {
	var err error
	if 1 == 2 {
	}

	if nil != err {
	} // want "lhs nil"

	if 1 == 2 ||
		err == nil ||
		nil == err {
	} // want "lhs nil"

	if nil == err && // want "lhs nil"
		nil == bar() {
	} // want "lhs nil"

	if 1 == 2 {
	} else if nil == err {
	} // want "lhs nil"

	if 1 == 2 || (2 == 3 && nil == err) {
	} // want "lhs nil"

	if nil != bar() {
	} // want "lhs nil"

	if nil != func() error { // want "lhs nil"
		return nil
	}() {
	}
}

func bar() error {
	return nil
}
