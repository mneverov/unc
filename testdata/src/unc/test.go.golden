package unc

func foo() {
	var err error
	if 1 == 2 {
	}

	if err != nil {
	} // want "lhs nil"

	if 1 == 2 ||
		err == nil ||
		err == nil {
	} // want "lhs nil"

	if err == nil && // want "lhs nil"
		bar() == nil {
	} // want "lhs nil"

	if 1 == 2 {
	} else if err == nil {
	} // want "lhs nil"

	if 1 == 2 || (2 == 3 && err == nil) {
	} // want "lhs nil"

	if bar() != nil {
	} // want "lhs nil"

	if func() error {
		return nil
	}() != nil {
	}
}

func bar() error {
	return nil
}